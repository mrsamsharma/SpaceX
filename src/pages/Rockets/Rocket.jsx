import React from 'react';
import { bookRockets, cancelBooking } from '../../redux/Rockets/RocketsSlice';
import { useDispatch } from 'react-redux';

const Rocket = (props) => {
  const rocket = props;
  const dispatch = useDispatch();

  const bookRocket = (id) => {
    dispatch(bookRockets(id));
  }

  const cancelRocket = (id) => {
    dispatch(cancelBooking(id));
  }

 return (
   <section className='container text-center rocket-container'>
     <div className='row justify-content-center'>
       <div className='col-3'>
         <img
           src={rocket.flickr_images}
           alt='rocket photo'
           className='rocket-img'
         />
       </div>
       <div className='col-6 info'>
         <h1>{rocket.rocket_name}</h1>
         <p>
           {rocket.reserved && (
             <span className='btn btn-primary btn-sm m-1'>Reserved</span>
           )}
           {rocket.description}
         </p>
         {rocket.reserved ? (
           <button
             onClick={() => cancelRocket(rocket.id)}
             className='btn btn-light btn-lg btn-outline-dark'
           >
             Cancel Reservation
           </button>
         ) : (
           <button
             onClick={() => bookRocket(rocket.id)}
             className='btn btn-primary btn-lg'
           >
             Reserve Rocket
           </button>
         )}
       </div>
     </div>
   </section>
 );
}
 
export default Rocket;