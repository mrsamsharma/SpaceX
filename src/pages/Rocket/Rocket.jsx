import React from 'react';
import { useDispatch } from 'react-redux';
import { bookRockets, cancelBooking } from '../../redux/Rockets/Rockets';

const Rocket = (props) => {
  const rocket = props;
  const dispatch = useDispatch();

  const bookRocket = (id) => {
    console.log(id);
    dispatch(bookRockets(id));
  };

  const cancelRocket = (id) => {
    console.log(id);
    dispatch(cancelBooking(id));
  };

  return (
    <section className='container text-center rocket-container'>
      <div className='col-4'>
        <img
          src={rocket.flickr_images}
          alt='rocket images'
          className='rocket-img'
        />
      </div>
      <div className='col-6 info'>
        <h1>{rocket.rocket_name}</h1>
        <p>
          {rocket.reserved && (
            <p className='btn btn-primary btn-sm m-1'>Reserved</p>
          )}
          {rocket.description}
        </p>
        {rocket.reserved ? (
          <button
            type='button'
            className='btn btn-light btn-lg btn-outline-dark'
            onClick={() => cancelRocket(rocket.id)}
          >
            Cancel Reservation
          </button>
        ) : (
          <button
            type='button'
            onClick={() => bookRocket(rocket.id)}
            className='btn btn-primary btn-lg'
          >
            Reserve Rocket
          </button>
        )}
      </div>
    </section>
  );
};
export default Rocket;
