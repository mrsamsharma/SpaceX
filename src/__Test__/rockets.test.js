import { render } from "@testing-library/react";
import React from "react";
import { Provider } from 'react-redux';
import store from '../redux/ConfigStore';
import Rocket from '../pages/Rocket/Rocket';
import RocketsContainer from '../pages/Rocket/Rockets';

describe('testing Rocket component', () => {
 it('should render the rocket component', () => {
  const rocket = render(
   <Provider store={store}>
   <Rocket />
   </Provider>
  );
  expect(rocket).toMatchSnapshot();
 });
 it('should render the rockets component', () => {
  const rockets = render(
    <Provider store={store}>
      <RocketsContainer />
    </Provider>
  );
  expect(rockets).toMatchSnapshot();
 });
})