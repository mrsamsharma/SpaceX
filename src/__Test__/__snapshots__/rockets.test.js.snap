// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Rocket component should render the rocket component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="container text-center rocket-container"
      >
        <div
          class="col-4"
        >
          <img
            alt="rocket images"
            class="rocket-img"
          />
        </div>
        <div
          class="col-6 info"
        >
          <h1 />
          <p />
          <button
            class="btn btn-primary btn-lg"
            type="button"
          >
            Reserve Rocket
          </button>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="container text-center rocket-container"
    >
      <div
        class="col-4"
      >
        <img
          alt="rocket images"
          class="rocket-img"
        />
      </div>
      <div
        class="col-6 info"
      >
        <h1 />
        <p />
        <button
          class="btn btn-primary btn-lg"
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`testing Rocket component should render the rockets component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="container m-2"
      >
        <ul
          class="d-flex flex-column align-items"
        />
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="container m-2"
    >
      <ul
        class="d-flex flex-column align-items"
      />
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
